application/constants/templates.py

```diff
@@ -43,3 +49,51 @@ class HookTypes(StrEnum):
     Template application hook types.
     """
     kubernetes_job = 'kubernetes_job'
+
+
+class http(StrEnum):
+    """
+    Template HTTP component types.
+    """
+    url = 'url'
+    method = 'method'
+    headers = 'headers'
+    parameters = 'parameters'
+
+class httpMethods(StrEnum):
+    """
+    Template HTTP component methods.
+    """
+    get = 'GET'
+    post = 'POST'
+    put = 'PUT'
+    patch = 'PATCH'
+    delete = 'DELETE'
+
+
+class httpDeploy(BaseModel):
+    """
+    Deploy schema for HTTP type components.
+    """
+    url: constr(min_length=1, strip_whitespace=True) = Field(description='URL for HTTP deployment', example='http://example.com')
```

Comment:
Please use `HttpUrl` for proper url validation. See next example here: https://github.com/JovianX/Service-Hub/blob/c6e80eb9e6a74462ba14ae036252407d66e96982/application/api/v1/schemas/helm.py#L23